datafusion/sql/src/unparser/plan.rs

```diff
@@ -271,8 +273,39 @@ impl Unparser<'_> {
                     relation,
                 )
             }
-            LogicalPlan::Distinct(_distinct) => {
-                not_impl_err!("Unsupported operator: {plan:?}")
+            LogicalPlan::Distinct(distinct) => {
+                let (select_distinct, input) = match distinct {
+                    Distinct::All(input) => (ast::Distinct::Distinct, input.as_ref()),
+                    Distinct::On(on) => {
+                        let exprs = on
+                            .on_expr
+                            .iter()
+                            .map(|e| self.expr_to_sql(e))
+                            .collect::<Result<Vec<_>>>()?;
+                        let items = on
+                            .select_expr
+                            .iter()
+                            .map(|e| self.select_item_to_sql(e))
+                            .collect::<Result<Vec<_>>>()?;
+                        match &on.sort_expr {
+                            Some(sort_expr) => {
+                                if let Some(query_ref) = query {
+                                    query_ref
+                                        .order_by(self.sort_to_sql(sort_expr.clone())?);
+                                } else {
+                                    return internal_err!(
+                                "Sort operator only valid in a statement context."
+                            );
```

Comment:
When implementing order by in a distinct on.
I found the order by should be added to `query` instead of `select`.
So my question is, what's the usage of `sort_by` in  `SelectBuilder`? I searched the code but not found an usage. 🤔

Reply:
I think it is to support the postgres style `DISTINCT ON`, which may not have all the columns of the select list

Perhaps @gruuya  knows more

Reply:
The `order_by`/`sort_by` clause is fetched from the `sqlparser::ast::Query` struct while parsing the `DISTINCT ON` query, so it makes sense that it is added there when de-parsing
https://github.com/apache/datafusion/blob/3dc1773403532752db01da5c545b2bdab415e489/datafusion/sql/src/query.rs#L132-L136

Indeed `sqlparser::ast::Select`s `sort_by` is not being used in DataFusion
https://github.com/apache/datafusion/blob/3dc1773403532752db01da5c545b2bdab415e489/datafusion/sql/src/select.rs#L68-L70
so it also seems that  `SelectBuilder::sort_by` is not needed atm.

Reply:
> Do you think it's better to remove `SelectBuilder::sort_by` or keep it?

If you're asking me I think it's fine to keep it (though other projects wouldn't be using it as well since `SelectBuilder` is `pub(super)`, so it wouldn't hurt to remove it either).